frontend/routes/components/hoverableUsername/hoverableUsername.component.tsx

```diff
@@ -0,0 +1,56 @@
+/**
+ *  Copyright (C) 2017 3D Repo Ltd
+ *
+ *  This program is free software: you can redistribute it and/or modify
+ *  it under the terms of the GNU Affero General Public License as
+ *  published by the Free Software Foundation, either version 3 of the
+ *  License, or (at your option) any later version.
+ *
+ *  This program is distributed in the hope that it will be useful,
+ *  but WITHOUT ANY WARRANTY; without even the implied warranty of
+ *  MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE.  See the
+ *  GNU Affero General Public License for more details.
+ *
+ *  You should have received a copy of the GNU Affero General Public License
+ *  along with this program.  If not, see <http://www.gnu.org/licenses/>.
+ */
+
+import * as React from 'react';
+import Tooltip from '@material-ui/core/Tooltip';
+import { Container, Username, TooltipText, FullName, CompanyName } from './hoverableUsername.styles';
+
+interface IProps {
+	name: string;
+	teamspace: string;
+	fetchUserDetails: (teamspace, username) => void;
+	userDetails: any;
+}
+
+export class HoverableUsername extends React.PureComponent<IProps, any> {
```

Comment:
I think it should be something like `DynamicUsername` because the name of the component will have to change when kind of interaction had changed. ;)